native/data_dypes.md

```diff
@@ -0,0 +1,102 @@
+### Abstracting Data Types
+
+An initial effort to abstract the data types between JS engine(s) and addon(s).
+
+#### Definitions: (Alphabetical order)
+
+ - Addon or Native Addon : Node.JS modules developed using C,C++.
+ - Chakra : A Javascript engine developed by Microsoft
+ - ECMAScript : The scripting language standardized by Ecma International in the ECMA-262 specification and ISO/IEC 16262. (ref Wikipedia)
+ - IO.JS : [Bringing ES6 to the Node Community!](http://iojs.org)
+ - JS Engine : Javascript engine that powers Node.JS or it's forks (Chakra, SpiderMonkey, V8 ...)
+ - JXcore : [Node.JS on mobile and embedded devices](http://jxcore.io/)
+ - MDN : Mozilla Developers Network
+ - Node : [Node.JS](http://nodejs.org) and it's previously mentioned sub projects
+ - SpiderMonkey : A Javascript engine developed by Mozilla.
+ - V8 : A Javascript engine developed by Google.
+
+
+#### JS Types
+
+The latest ECMAScript standard defines seven data types: (ref. MDN and contributors [2015] [Online] Available from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures))
+
+    Six data types that are primitives:
+        Boolean
+        Null
+        Undefined
+        Number
+        String
+        Symbol (new in ECMAScript 6)
+    and 
+        Object
+
+Initial list of the corresponding native types are given below;
+
+```
+enum JSTypes {
+  Int32, // Number
```

Comment:
AFAIK UInt32 is just a static cast from number type. Int32 and Double are the only real types backing them.